src/metadatacache/MetadataCache.ts

```diff
@@ -338,19 +207,39 @@ export class MetadataCache {
     return `${this.url}`
   }
 
-  private transformResult(
-    { results, page, total_pages: totalPages, total_results: totalResults }: any = {
-      result: [],
-      page: 0,
-      total_pages: 0,
-      total_results: 0
-    }
-  ): QueryResult {
-    return {
-      results: (results || []).map((ddo: DDO) => new DDO(ddo as DDO)),
-      page,
-      totalPages,
-      totalResults
-    }
+  /**
+   * Simple blocking sleep function
+   */
+  public sleep(ms: number) {
+    return new Promise((resolve) => {
+      setTimeout(resolve, ms)
+    })
+  }
+
+  /**
+   * Blocks until Aqua will cache the did (or the update for that did) or timeouts
+   * @param  {string} did DID of the asset.
+   * @param  {string} txid used when the did exists and we expect an update with that txid.
+   * @return {Promise<DDO>} DDO of the asset.
+   */
+  public async waitForAqua(did: string, txid?: string) {
```

Comment:
why do you have this here? looks like a function used just in testing.

Reply:
we could use waitForAqua in the market as well, when publishing, instead of just doing a pause..